sqlalchemy_redshift/dialect.py

```diff
@@ -641,23 +643,85 @@ def keyfunc(column):
             'redshift_interleaved_sortkey': interleaved_sortkey,
         }
 
+    def _do_generate_iam_auth_token(self, redshift, cparams):
+        # Hostname is the cluster identifier
+        cluster_id = cparams['host']
+        response = redshift.describe_clusters(ClusterIdentifier=cluster_id)
+        # If there is no cluster, it will throw an exception
+        rs_cluster = response['Clusters'][0]
+
+        response = redshift.get_cluster_credentials(
+            DbUser=cparams['user'],
+            DbName=cparams['database'],
+            ClusterIdentifier=cluster_id,
+            AutoCreate=False)
+
+        # Extract user, password, hostname, and port from the response
+        return {
+            'host': rs_cluster['Endpoint']['Address'],
+            'port': rs_cluster['Endpoint']['Port'],
+            'user': response['DbUser'],
+            'password': response['DbPassword'],
+            'database': cparams['database']
+        }
+
+    def _generate_iam_auth_token(self, cparams):
+        """
+        Parses for necessary information to generate username and
+        password using the necessary Redshift API.
+
+        Username and database are passed to the get_cluster_credentials()
+        API to verify permissions.
+        """
+        try:
+            import boto3
+            if self.aws_region not in cparams:
+                raise ValueError(
+                    "Need AWS Region as part of the connection configuration.")
+
+            redshift = boto3.client('redshift',
+                                    region_name=cparams[self.aws_region])
+
+            # Verify that the parameters make sense.
+            if "password" in cparams:
+                raise ValueError(
+                    "'password' cannot be used when connecting with"
+                    " IAM federation."
+                )
+
+            return self._do_generate_iam_auth_token(redshift, cparams)
+
+        except ImportError:
+            raise ImportError(
```

Comment:
unreachable

Reply:
I wanted to check if it's fine to add boto3 as a dependency, IMHO it makes sense because Redshift is at the end of the day a AWS product :)